### ISSUE TITLE
dhall-to-json None value not supported for optionals.

### ISSUE THREAD

**Issue by andrewchambers** [dhall-lang/dhall-lang#492]
```
 $ dhall-to-json <<< "Some 1"
1
$  dhall-to-json <<< "None"
Error: Cannot translate to JSON
```

**Comment by andrewchambers** [comment]
Not exactly sure what the best approach is, but it would be nice for my go package that uses dhall-to-json to be able to use optionals.

**Comment by singpolyma** [comment]
> $ dhall-to-json <<< "None"

I believe you want something like `None Natural` -- the type is required.

**Comment by andrewchambers** [comment]
My mistake :)

**Comment by Profpatsch** [comment]
@singpolyma The way `None` works in the AST, you could also make `None` valid by itself, no?

**Comment by singpolyma** [comment]
‎> @singpolyma The way `None` works in the AST, you could also make `None` valid by itself, no?

By itself it is valid -- it's a function from a type to the empty optional of that type.‎

**Comment by Profpatsch** [comment]
> By itself it is valid -- it's a function from a type to the empty optional of that type.‎

I know, but the AST doesn’t express that fact, so you could re-interpret it. Admittedly that would be a crude hack and go against dhall semantics.

**Comment by Gabriel439** [comment]
I don't think we should translate a bare `None` to `null`, but a better error message would be very easy to add.  I will reopen this and make that the scope of this issue

**Comment by Gabriel439** [comment]
The fix is up here: https://github.com/dhall-lang/dhall-haskell/pull/895
